xetldfs/src/xet_interface.rs

```diff
@@ -1,194 +1,163 @@
-use std::sync::{Arc, RwLock};
-use tokio::sync::Mutex as TMutex;
-
-use libxet::config::XetConfig;
-use libxet::data::{PointerFile, PointerFileTranslatorV2};
-use libxet::errors::Result;
-use libxet::git_integration::{get_repo_path, GitXetRepo};
-use libxet::ErrorPrinter;
-
-use crate::path_utils::{path_of_fd, resolve_path};
-use crate::runtime::{tokio_run, with_interposing_disabled};
+use path_absolutize::Absolutize;
+use std::collections::HashMap;
+use std::os::unix::ffi::OsStrExt;
+use std::path::Path;
+use std::sync::Arc;
+
+use crate::path_utils::{absolute_path, absolute_path_c, path_of_fd};
+use crate::real_close;
+use crate::repo_path_utils::verify_path_is_git_repo;
+use crate::runtime::{self, with_interposing_disabled};
+use crate::utils::cstring_to_str;
+use crate::xet_repo_wrapper::XetFSRepoWrapper;
 use crate::xet_rfile::XetFdReadHandle;
-use crate::{cstring_to_str, real_close};
 use crate::{ld_trace, ld_warn};
 use crate::{my_dup2, real_open};
 
+use libc::{c_char, c_int};
+
+// Constructing and storing the repository paths
 lazy_static! {
-    static ref XET_REPO_WRAPPERS: RwLock<Vec<Arc<XetFSRepoWrapper>>> = RwLock::new(Vec::new());
-    static ref XET_ENVIRONMENT_CFG: TMutex<Option<XetConfig>> = TMutex::new(None);
+    static ref XET_REPOS: Vec<Arc<XetFSRepoWrapper>> = initialize_repo_paths();
 }
 
-pub static XET_LOGGING_INITIALIZED: AtomicBool = AtomicBool::new(false);
-
-// Requires runnig inside tokio runtime, so async
-async fn get_base_config() -> Result<XetConfig> {
-    // If the base config isn't set, then initialize everthing.
-    let mut cfg_wrap = XET_ENVIRONMENT_CFG.lock().await;
-
-    if cfg_wrap.is_none() {
-        let cfg = XetConfig::new(None, None, libxet::config::ConfigGitPathOption::NoPath)?;
+/// Loads the repository paths from the environment variable XET_LDFS_REPO.
+///
+/// This should be a ;-separated list of one or more absolute paths to a repository.
+fn initialize_repo_paths() -> Vec<Arc<XetFSRepoWrapper>> {
+    let _ig = with_interposing_disabled();
 
-        libxet::environment::log::initialize_tracing_subscriber(&cfg)?;
+    // Get the environment variable
+    let repo_env = std::env::var("XET_LDFS_REPO").unwrap_or_default();
 
-        // Error reporting is initialized.
-        XET_LOGGING_INITIALIZED.store(true, std::sync::atomic::Ordering::SeqCst);
+    ld_trace!("initialize_repo_paths: XET_LDFS_REPO={repo_env}");
 
-        let _ = openssl_probe::try_init_ssl_cert_env_vars();
+    // Split the environment variable by ':'
+    let paths: Vec<&str> = repo_env.split(':').collect();
 
-        ld_trace!("CFG initialized.");
+    // Initialize the vector to hold the results
+    let mut repos: Vec<Arc<XetFSRepoWrapper>> = Vec::with_capacity(paths.len());
 
-        *cfg_wrap = Some(cfg);
-    }
+    for mut path in paths {
+        // Trim out the whitespace.
+        path = path.trim();
 
-    Ok(cfg_wrap.as_ref().unwrap().clone())
-}
+        // Is the path quoted?
+        if path.len() >= 2
```

Comment:
Just realized that *unix allow all special characters in a path component except for `/`, this means that `"a.txt"` is actually a valid path name. And since shells remove quotes typed in the command line unless users deliberately want them in by escaping the quotes, we should treat them as users' intention.

Reply:
Ah, gotcha, good point.  I guess the user would quote the entire environment variable string in this case.